src/ipa_pc/mod.rs

```diff
@@ -744,55 +784,81 @@ impl<G: AffineCurve, D: Digest> PolynomialCommitment<G::ScalarField> for InnerPr
                 }
                 combined_commitment_proj += &comm_lc.mul(&cur_challenge);
 
-            } else {
+                if hiding_bound.is_some() {
```

Comment:
Why not combining the blinding randomnesses where we combine the segment commitments, i.e. in the loop above?

Reply:
Yes, thank you, I've combined multiple loops into one